packages/py/polywrap-msgpack/tests/test_msgpack.py

```diff
@@ -71,3 +69,119 @@ def test_generic_map_decode():
+
+# COMPLEX NUMBERS
+
+def test_sanitize_complex_number_returns_string():
+    assert sanitize(3 + 5j) == "(3+5j)"
+    assert sanitize(0 + 9j) == "9j"
+
+def test_sanitize_simple_dict_returns_sanitized_values(simple_dict):
+    assert sanitize(simple_dict) == simple_dict
+
+# SLOTS
+
+def test_sanitize_object_with_slots_attributes_returns_dict_instead(object_with_slots_attributes, object_with_slots_sanitized):
+    assert sanitize(object_with_slots_attributes) == object_with_slots_sanitized
+
+# TUPLES
+
+def test_sanitize_single_tuple_returns_list(single_tuple):
+    # To create a tuple with only one item, you have add a comma after the item, 
+    # otherwise Python will not recognize the variable as a tuple.
+    assert type(sanitize(single_tuple)) == list
+    assert sanitize(single_tuple) == [8] 
+
+def test_sanitize_long_tuple_returns_list():
+    assert sanitize((2,3,6)) == [2,3,6] 
+
+def test_sanitize_nested_tuples_returns_nested_list(nested_tuple, nested_list):
+    assert sanitize(nested_tuple) == nested_list
+
+
+# SETS
+
+def test_sanitize_set_returns_list(set1):
+    # Remember sets automatically reorganize the contents of the object
+    # meaning {'bob', 'alice'} might be stored as {'alice','bob'} in memory
+    assert type(sanitize(set1)) == list
+
+def test_sanitize_set_returns_list_with_all_items_of_the_set(set1, set2):
+    sanitized = sanitize(set1)
+    r = []       
+    [r.append(True) if item in sanitized else r.append(False) for item in set1]
+    assert False not in r
+    
+    sanitized = sanitize(set2)
+    r = []       
+    [r.append(True) if item in sanitized else r.append(False) for item in set2]
+    assert False not in r
+
+def test_sanitize_set_returns_list_of_same_length(set1):
+    assert len(sanitize(set1)) == len(set1)
+
+
+def test_sanitize_complex_dict_returns_sanitized_values():
+    # This test is not passing because when we sanitize a set, the contents are shuffled around.
+    complex_dict = {'name': ['John', 'Doe'],
+        'position':[-0.34478,12.98453],
+        'color': 'green',
+        'age':33,
+        'origin':(0,0),
+        'is_online': True,
+        'pet': None,
+        'friends': {'bob','alice','megan','john'} }
+    sanitized_complex_dict = {'name': ['John', 'Doe'],
+        'position':[-0.34478,12.98453],
+        'color': 'green',
+        'age':33,
+        'origin':[0,0],
+        'is_online': True,
+        'pet': None,
+        'friends': ['bob','alice','megan','john'] }
+    assert sanitize(complex_dict) == sanitized_complex_dict
+
+# DATA CLASSES
+
+def test_sanitize_dataclass_object_returns_dict(dataclass_object1, dataclass_object1_as_dict):
+    assert sanitize(dataclass_object1) == dataclass_object1_as_dict
+
+def test_sanitize_list_of_dataclass_objects_returns_list_of_dicts(dataclass_object1, dataclass_object2):
+    print(dataclass_object1)
+    assert sanitize([dataclass_object1, dataclass_object2]) == [dataclass_object1.__dict__, dataclass_object2.__dict__]
+
```

Comment:
test dict of dataclass returns dict of dict

Reply:
added 2 new tests for this case, one for normal dataclass objects and other one without.